eth/tracers/internal/tracetest/testdata/prestate_tracer_with_diff_mode/create_failed.json

```diff
@@ -73,7 +73,7 @@
   "result": {
     "pre": {
       "0x808b4da0be6c9512e948521452227efc619bea52": {
-        "balance": "0x2cdb96c56db040b43",
+        "balance": "0x2cddf395ac4f56170",
```

Comment:
Why is the pre balance changed?

Reply:
Good question. I din't think to much about it because the difference matched the expected value. Could that be a bug in the tracing code?

Reply:
The genesis alloc:
```
      "0x808b4da0be6c9512e948521452227efc619bea52": {
        "balance": "0x2cdb96c56db040b43",
        "nonce": "1223932"
```
So this PR makes it wrong, whereas it was correct earlier.

Reply:
Yeah I'm not sure why you changed pre balance, that doesn't need to be. I had actually fixed this issue in https://github.com/ethereum/go-ethereum/pull/27629/files#diff-13c76087ba15a5cdb7f3003416a918dd7269b3fac232fc98908d8be927fed6d9. If you revert the pre-balance then you should get post balance of: `0x2cd987071ba2346b6`.

Reply:
Yes, you're both right that the pre-balance shouldn't change. I dug a bit deeper and that seems to be another issue in how the test is setup.

In the prestate tracer, the sender balance is adjusted to incorporate the transaction cost: https://github.com/palango/go-ethereum/blob/1537ce63280daa92914bfdd3a073c3bca2c6e500/eth/tracers/native/prestate.go#L109-L112

The problem here is that it uses `env.TxContext.GasPrice`, which doesn't take care of the base fee: https://github.com/palango/go-ethereum/blob/1537ce63280daa92914bfdd3a073c3bca2c6e500/eth/tracers/internal/tracetest/prestate_test.go#L99

Is there an elegant way to fix this without making the explicit calculation in there?

Reply:
I'm pushing a fix to this as we speak.

Reply:
@s1na I also found a simple way, just pushed a commit.

Reply:
Looks good. Please apply it to the other test files as well.

Reply:
> Looks good. Please apply it to the other test files as well.

Done. And sorry for overwriting your commit.